File: ios/LocationTrackeriOS/LocationTrackeriOS/Config/ApiConfig.swift
import Foundation

enum ApiConfig {
    static let apiBaseURL = URL(string: "https://beapis-in.staging.geoiq.ai/retailapp/stg/v3/")!
    static let locationUploadURL = URL(string: "https://beapis-in.staging.geoiq.ai/bdapp/stg/v1/bd/locationTracking/updateUserLocation")!
    static let locationUploadBatchURL = URL(string: "https://beapis-in.staging.geoiq.ai/bdapp/stg/v1/bd/locationTracking/updateUserLocationBatch")!
    static let loginPath = "user/userlogin"

    // Force-update version check
    // Returns { "version": "1.0", "version_code": 1, "status": 200 }
    static let appVersionCheckURL = URL(string: "https://beapis-in.staging.geoiq.ai/retailapp/stg/v3/app/getAppConfig?app=location-tracker&platform=ios")!

    // App Store link — opened when user taps "Update Now" on the force update screen
    // Replace with a direct APK/IPA URL to distribute outside the App Store
    static let appStoreURL = URL(string: "https://apps.apple.com/app/id<YOUR_APP_ID>")!
}

File: app/src/main/java/com/example/locationtracker/data/network/ApiConfig.kt
package com.example.locationtracker.data.network

object ApiConfig {
    // Equivalent to EXPO_PUBLIC_API_BASE_URL
    const val API_BASE_URL = "https://beapis-in.staging.geoiq.ai/retailapp/stg/v3/"

    // Equivalent to EXPO_PUBLIC_LOCATION_UPLOAD_URL (full URL)
    const val LOCATION_UPLOAD_URL =
        "https://beapis-in.staging.geoiq.ai/bdapp/stg/v1/bd/locationTracking/updateUserLocation"
    const val LOCATION_UPLOAD_BATCH_URL =
        "https://beapis-in.staging.geoiq.ai/bdapp/stg/v1/bd/locationTracking/updateUserLocationBatch"

    const val LOGIN_PATH = "user/userlogin"

    // Config only (currently unused)
    const val FORGOT_PASSWORD_PATH = "user/forgot-password"
    const val RESET_PASSWORD_PATH = "user/reset-password"
}
